packages/nx-plugin/src/plugin/plugin.unit.test.ts

```diff
@@ -0,0 +1,66 @@
+import { CreateNodesContext } from '@nx/devkit';
+import { vol } from 'memfs';
+import { describe, expect } from 'vitest';
+import { createFilesAndInvokeCreateNodesOnThem } from '@code-pushup/test-utils';
+import { PROJECT_JSON_FILE_NAME } from '../internal/constants';
+import { CP_TARGET_NAME } from './constants';
+import { createNodes } from './plugin';
+
+describe('@code-pushup/nx-plugin/plugin', () => {
+  let context: CreateNodesContext;
+
+  beforeEach(() => {
+    context = {
+      nxJsonConfiguration: {},
+      workspaceRoot: '',
+    };
+  });
+
+  afterEach(() => {
+    vol.reset();
+  });
+
+  it('should normalize context and use it to create target on ROOT project', async () => {
+    const projectRoot = '.';
+    const matchingFilesData = {
+      [`${projectRoot}/${PROJECT_JSON_FILE_NAME}`]: `${JSON.stringify({
+        name: '@org/empty-root',
+      })}`,
+    };
+    const projects = await createFilesAndInvokeCreateNodesOnThem(
+      createNodes,
+      context,
+      {},
+      { matchingFilesData },
+    );
+
+    // project should have one target created
+    const targets = projects[projectRoot]?.targets ?? {};
+    expect(Object.keys(targets)).toHaveLength(1);
+
+    // target should be the configuration target
+    expect(targets[`${CP_TARGET_NAME}--configuration`]).toBeDefined();
+  });
+
+  it('should normalize context and use it to create target on PACKAGE project', async () => {
+    const projectRoot = 'apps/my-app';
+    const matchingFilesData = {
+      [`${projectRoot}/${PROJECT_JSON_FILE_NAME}`]: `${JSON.stringify({
+        name: '@org/empty-root',
+      })}`,
+    };
+    const projects = await createFilesAndInvokeCreateNodesOnThem(
+      createNodes,
+      context,
+      {},
+      { matchingFilesData },
+    );
+
+    // project should have one target created
```

Comment:
I recommend having one direct assertion regarding the content of the targets instead of 3. This would also get rid of the nullish operator causing branching logic.

Applicable in multiple places.

Reply:
i don't get it

Reply:
I think she means something like this?

![image](https://github.com/user-attachments/assets/39d09e70-903c-4393-a241-b4c3df51701d)

Reply:
@matejchalk Yes, exactly! Thank you, I was on my tablet when I gave you the review, not the best place to write code 😅

Reply:
Yesterday night it popped up again. And I remembered previose similar feedback.. Will do! THX!

Reply:
Updated now!  Thx for the hint, WAY better now.